utility_code/lib_py3/common.py

```diff
@@ -22,19 +25,41 @@ def get_entity_name_from_nbt(entity_nbt: nbt.TagCompound, remove_color=True) ->
         return None
     return parse_name_possibly_json(entity_nbt.at_path('CustomName').value, remove_color)
 
-def get_item_name_from_nbt(item_nbt: nbt.TagCompound, remove_color=True):
+def get_item_name_from_nbt(item_tag: nbt.TagCompound, remove_color=True, include_masterwork_level=False) -> str:
     """
     Parses a color-removed name out of an item's NBT. Returns a string or None if no name exists
+    if include_masterwork_level is True, _m{masterwork_level} will be appended to the item's name if both name and masterwork level exist
     """
-    if not item_nbt.has_path("display.Name"):
-        if item_nbt.has_path("title"):
-            title = item_nbt.at_path("title").value
+    if not item_tag.has_path("display.Name"):
+        if item_tag.has_path("title"):
+            title = item_tag.at_path("title").value
             if remove_color:
                 title = unformat_text(title)
             return title
         return None
 
-    return parse_name_possibly_json(item_nbt.at_path("display.Name").value, remove_color)
+    item_name = parse_name_possibly_json(item_tag.at_path("display.Name").value, remove_color)
+    if include_masterwork_level:
+        masterwork_level = get_masterwork_level_from_nbt(item_tag)
+        if masterwork_level is not None:
+            item_name = f"{item_name}_m{masterwork_level}"
+
+    return item_name
+
+# Returns
+def get_masterwork_level_from_nbt(item_tag: nbt.TagCompound, err_print_on_inval=True) -> str:
+    """
+    Gets the masterwork level from the item's NBT. Returns the string level if present, otherwise None
+    If the value is invalid it is still returned, and an error is printed to stderr if err_print_on_inval is True
+    """
+    if item_tag.has_path('Monumenta.Masterwork'):
+        masterwork_level = item_tag.at_path('Monumenta.Masterwork').value
+        if err_print_on_inval and masterwork_level not in ['0', '1', '2', '3', '4', '5', '6', '7a', '7b', '7c']:
```

Comment:
So, there is a "none" masterwork level and an "error" masterwork level that are pretty much just there for debugging but they're there nonetheless, should probably add separate catch for those two cases

Reply:
Fixed, thanks